helix-core/src/surround.rs

```diff
@@ -39,13 +39,98 @@ pub fn find_nth_pairs_pos(
     n: usize,
 ) -> Option<(usize, usize)> {
     let (open, close) = get_pair(ch);
-    // find_nth* do not consider current character; +1/-1 to include them
-    let open_pos = search::find_nth_prev(text, open, pos + 1, n, true)?;
-    let close_pos = search::find_nth_next(text, close, pos - 1, n, true)?;
+
+    let (open_pos, close_pos);
+    if open == close {
+        // find_nth* do not consider current character; +1/-1 to include them
+        open_pos = search::find_nth_prev(text, open, pos + 1, n, true)?;
+        close_pos = search::find_nth_next(text, close, pos - 1, n, true)?;
+    } else {
+        open_pos = find_nth_open_pair(text, open, close, pos, n)?;
+        close_pos = find_nth_close_pair(text, open, close, pos, n)?;
+    }
```

Comment:
You could do
```suggestion
    let (open_pos, close_pos) = if open == close {
        // find_nth* do not consider current character; +1/-1 to include them
        (
            search::find_nth_prev(text, open, pos + 1, n, true)?,
            search::find_nth_next(text, close, pos - 1, n, true)?,
        )
    } else {
        (
            find_nth_open_pair(text, open, close, pos, n)?,
            find_nth_close_pair(text, open, close, pos, n)?,
        )
    };
```
But it's a matter of preference.

Reply:
Done 👍🏾